nixos/modules/services/web-apps/cryptpad.nix

```diff
@@ -0,0 +1,447 @@
+          };
+          websocketPort = mkOption {
+            type = types.int;
+            default = 3003;
+            description = mdDoc "Backend websocket port";
+          };
+          maxWorkers = mkOption {
+            type = types.nullOr types.int;
+            default = null;
+            description = mdDoc "Number of child processes, defaults to number of cores available";
+          };
+          adminKeys = mkOption {
+            type = types.listOf types.str;
+            default = [];
+            description = mdDoc "Admins allowed to admin panel. Keys can be found in the settings page of the user";
+            example = [ "[cryptpad-user1@my.awesome.website/YZgXQxKR0Rcb6r6CmxHPdAGLVludrAF2lEnkbx1vVOo=]" ];
+          };
+          logToStdout = mkOption {
+            type = types.bool;
+            default = true;
+            description = mdDoc "Whether to log to stdout (will go to systemd service). Level can be changed with logLevel.";
+          };
+          blockDailyCheck = mkOption {
+            type = types.bool;
+            default = false;
+            description = mdDoc "Disable telemetry. This has priority over the 'Server telemetry' setting in admin panel.";
+          };
+          installMethod = mkOption {
+            type = types.str;
+            default = "nixos";
+            description = mdDoc ''
+              Install method is listed in telemetry if you agree to it through the consentToContact
+              setting in the admin panel.
+            '';
+          };
+        };
+      };
+    };
+  };
+
+  config = mkIf cfg.enable (mkMerge [
+    {
+      users.users.cryptpad = {
+        isSystemUser = true;
+        group = "cryptpad";
+      };
+      users.groups.cryptpad = {};
+
+      systemd.services.cryptpad = {
+        description = "Cryptpad service";
+        wantedBy = [ "multi-user.target" ];
+        after = [ "networking.target" ];
+        serviceConfig = {
+          User = "cryptpad";
+          Environment = [
+            "CRYPTPAD_CONFIG=${conf_file}"
+            "HOME=%S/cryptpad"
+          ];
+          ExecStart = "${cfg.package}/bin/cryptpad";
+          PrivateTmp = true;
+          Restart = "always";
+          StateDirectory = "cryptpad";
+          WorkingDirectory = "%S/cryptpad";
+        };
+      };
+    }
+    (mkIf cfg.confinement {
+      systemd.services.cryptpad = {
+        serviceConfig.BindReadOnlyPaths = [
+          conf_file
+          # apparently needs proc for workers management
+          "/proc"
+          "/dev/urandom"
+        ] ++ (if ! cfg.config.blockDailyCheck then [
+          "/etc/resolv.conf"
+          "/etc/hosts"
+        ] else []);
+        confinement = {
+          enable = true;
+          binSh = null;
+          mode = "chroot-only";
+        };
+      };
+    })
+    (mkIf cfg.nginx.enable {
+      assertions = [
+        { assertion = nginx.enable;
+          message = "nginx must be enabled";
```

Comment:
IMHO, the way we do it in other modules is to have a portal from `cfg.nginx` inside a virtual host fragment of configuration.

This way, you can do things like `services.cryptpad.nginx.locations."/".xyz`, etc. and act upon the configuration from there.

It makes a lot of sense because then you can extend the NixOS module with any web server as long as you have the internal domain knowledge.

Conclusion: It should not enable NGINX, it should not assert NGINX is enabled. As it is in other modules.